src/Compilers/CSharp/Portable/Syntax/LookupPosition.cs

```diff
@@ -289,24 +289,9 @@ private static SyntaxToken GetFirstIncludedToken(StatementSyntax statement, bool
                 case SyntaxKind.FixedStatement:
                     return ((FixedStatementSyntax)statement).FixedKeyword;
                 case SyntaxKind.ForEachStatement:
-                    // NB: iteration variable is only in scope in body.
-                    ForEachStatementSyntax forEachSyntax = (ForEachStatementSyntax)statement;
-                    if (inRecursiveCall)
-                    {
-                        return forEachSyntax.ForEachKeyword;
-                    }
-                    return GetFirstIncludedToken(forEachSyntax.Statement, inRecursiveCall: true);
+                    return ((ForEachStatementSyntax)statement).OpenParenToken.GetNextToken();
                 case SyntaxKind.ForStatement:
-                    // Section 8.8.3 of the spec says that the scope of the loop variable starts at 
-                    // its declaration.  If it's not there, then the scope we are interested in is
-                    // the loop body.
-                    ForStatementSyntax forSyntax = (ForStatementSyntax)statement;
-                    if (inRecursiveCall)
-                    {
-                        return forSyntax.ForKeyword;
-                    }
-                    VariableDeclarationSyntax declOpt = forSyntax.Declaration;
-                    return declOpt == null ? GetFirstIncludedToken(forSyntax.Statement, inRecursiveCall: true) : declOpt.GetFirstToken();
+                    return ((ForStatementSyntax)statement).OpenParenToken.GetNextToken();
```

Comment:
In test file, all the combo tests (such as `ScopeOfPatternVariables_Do_01` and `ScopeOfPatternVariables_For_01`) are un-necessarily large and hard to review. It is painful to verify whether errors corresponding to each case is present and whether the error code is correct. This will also cause pain when any test breaks or needs to be updated (and diffed and reviewed again).